vllm/model_executor/models/paligemma.py

```diff
@@ -307,17 +307,19 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
                 if key_to_modify in name:
                     name = name.replace(key_to_modify, new_key)
             use_default_weight_loading = False
-            for (param_name, shard_name, shard_id) in stacked_params_mapping:
-                if shard_name not in name:
-                    continue
-                name = name.replace(shard_name, param_name)
-                # Skip loading extra bias for GPTQ models.
-                if name.endswith(".bias") and name not in params_dict:
-                    continue
-                param = params_dict[name]
-                weight_loader = param.weight_loader
-                weight_loader(param, loaded_weight, shard_id)
-                break
+            if "vision" not in name or self.vision_tower.shard_weight:
+                for (param_name, shard_name,
+                     shard_id) in stacked_params_mapping:
```

Comment:
@DarkLight1337 We use self.vision_tower.shard_weight here, since `Paligemma` doesn't use Siglip's `load_weight` function in its loading logic.

Reply:
Ahh I see now. Thanks for the explanation!